crates/sui-core/src/narwhal_manager/mod.rs

```diff
@@ -36,6 +40,31 @@ pub struct NarwhalConfiguration {
     pub registry_service: RegistryService,
 }
 
+pub struct NarwhalManagerMetrics {
+    start_latency: IntGaugeVec,
+    shutdown_latency: IntGauge,
+}
+
+impl NarwhalManagerMetrics {
+    pub fn new(registry: &Registry) -> Self {
+        Self {
+            start_latency: register_int_gauge_vec_with_registry!(
+                "narwhal_manager_start_latency",
+                "The latency of starting up narwhal nodes",
+                &["primary_retries", "worker_retries"],
```

Comment:
Since we are only recording latencies for the last restart, it seems putting primary_retries and worker_retries into separate gauges are better?

Reply:
Yeah I wanted to avoid adding specific metrics just for those since as you said they are practically tight to the last attempt - that's why I went with labels. But maybe they could deserve a separate metric - let me refactor